### PR TITLE
Xodr offset transformation

### PR DESCRIPTION
**Pull request by klemense1**

### CODE REVIEW

File: modules/geometry/line.hpp [hunk 1]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
   return bg::length<T>(line.obj_);
 }
 
+inline Line rotate(const Line &line, float hdg) {
+  namespace trans = boost::geometry::strategy::transform;
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: using boost::geometry::strategy::transform::rotate_transformer;

File: modules/geometry/line.hpp [hunk 2]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
...
+  trans::rotate_transformer<boost::geometry::radian, double, 2, 2> rotate(hdg);
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: rotate_transformer<boost::geometry::radian, double, 2, 2> rotate(hdg);

File: modules/geometry/line.hpp [hunk 3]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
   return bg::length<T>(line.obj_);
 }
 
+inline Line rotate(const Line &line, float hdg) {
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: inline Line rotate(const Line& line, float hdg) {

File: modules/geometry/line.hpp [hunk 4]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
...
+  namespace trans = boost::geometry::strategy::transform;
+  trans::rotate_transformer<boost::geometry::radian, double, 2, 2> rotate(hdg);
+  Line line_rotated;
+  boost::geometry::transform(line.obj_, line_rotated.obj_, rotate);
+  line_rotated.recompute_s();
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: The length(s) should not change due to a rotation?

File: modules/geometry/line.hpp [hunk 5]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
...
+  return line_rotated;
+}
+
+inline Line translate(const Line &line, float x, float y) {
+  namespace trans = boost::geometry::strategy::transform;
+  trans::translate_transformer<double, 2, 2> translate(x, y);
+  Line line_translated;
+  boost::geometry::transform(line.obj_, line_translated.obj_, translate);
+  line_translated.recompute_s();
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: Length(s) should not change due to translation.

File: modules/geometry/line.hpp [hunk 6]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
...
+  namespace trans = boost::geometry::strategy::transform;
+  trans::rotate_transformer<boost::geometry::radian, double, 2, 2> rotate(hdg);
+  Line line_rotated;
+  boost::geometry::transform(line.obj_, line_rotated.obj_, rotate);
+  line_rotated.recompute_s();
+  return line_rotated;
+}
+
+inline Line translate(const Line &line, float x, float y) {
+  namespace trans = boost::geometry::strategy::transform;
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: same as with the function above.

File: modules/geometry/line.hpp [hunk 7]
```diff
@@ -177,6 +177,24 @@ inline T length(const Line &line) {
...
+inline Line rotate(const Line &line, float hdg) {
+  namespace trans = boost::geometry::strategy::transform;
+  trans::rotate_transformer<boost::geometry::radian, double, 2, 2> rotate(hdg);
+  Line line_rotated;
+  boost::geometry::transform(line.obj_, line_rotated.obj_, rotate);
+  line_rotated.recompute_s();
+  return line_rotated;
+}
+
+inline Line translate(const Line &line, float x, float y) {
```
Comment (patrickhart) [inline modules/geometry/line.hpp]: inline Line translate(const Line& line, float x, float y) {

File: modules/world/opendrive/plan_view.cpp [hunk 8]
```diff
@@ -92,6 +92,14 @@ bool PlanView::add_arc(
   return true;
 }
 
+bool PlanView::apply_offset_transform(float x, float y, float hdg) {
+  geometry::Line rotated_line = rotate(reference_line_, hdg);
+  geometry::Line transformed_line = translate(rotated_line, x, y);
+  reference_line_ = transformed_line;
```
Comment (patrickhart) [inline modules/world/opendrive/plan_view.cpp]: reference_line_ = translate(rotated_line, x, y);

File: modules/geometry/line.hpp (hunk 4 above)
Comment (klemense1) [reply modules/geometry/line.hpp]: I know, just wanted to make sure ... I'll remove it

File: modules/runtime/commons/xodr_parser.py [hunk 9]
```diff
@@ -431,7 +452,7 @@ def convert_to_map(self, python_map):
       CPP Map -- Map for usage with CPP
     """
         for road in self.python_map["roads"]:
-            new_road = self.create_cpp_road(road)
+            new_road = self.create_cpp_road(road, self.python_map["header"])
```
Comment (patrickhart) [inline modules/runtime/commons/xodr_parser.py]: What happens if there is no header -- like in many files?

**Pull request merged**